satpy/readers/hsaf_h5.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019.
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""A reader for HDF5 Snow Cover (SC) file produced by the Hydrology SAF."""
+import logging
+import os
+from datetime import datetime
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+
+class HSAFFileHandler(BaseFileHandler):
+    """File handler for HSAF H5 files."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(HSAFFileHandler, self).__init__(filename,
+                                              filename_info,
+                                              filetype_info)
+
+        self._msg_datasets = {}
+        self._start_time = None
+        self._end_time = None
```

Comment:
Not needed, the `BaseFileHandler` class has the properties `start_time` and `end_time`. The `end_time` property needs to be overridden in this class, for example:
```python
    @property
    def end_time(self):
        """Get end time."""
        return self.filename_info['start_time'] + timedelta(hours=23, minutes=59, seconds=59)
```